org.gecko.emf.csv.tests/test.bndrun

```diff
@@ -5,6 +5,15 @@
 
 -runbundles.junit5: ${test.runbundles}
 
+-runrepos: \
```

Comment:
I can't add a comment to the line in the file, I want to so I add it here:

The Test if the Requirements work correctly would be, that the only runrequirement defined here is the `org.gecko.emf.csv`.tests. As it references the `EMFCSVResource`, everything required should be resolved by itself.

Reply:
This was tested already via `/org.gecko.emf.util.jakartars.bnd.library.workspace/required.bndrun` and `/org.gecko.emf.util.jaxrs.bnd.library.workspace/required.bndrun` and worked with no problems.

However, the test which you suggested does not work. What happens is a non-deterministic behavior where additional, unexpected exporter bundles are pulled in during resolve, i.e. when resolving via `/org.gecko.emf.csv.tests/test.bndrun`, having defined only `org.gecko.emf.csv.tests` in its run requirements, in addition to all of the expected bundles (`org.gecko.emf.csv`, `org.gecko.emf.csv.tests`, `org.gecko.emf.exporter`, `org.gecko.emf.exporter.csv`, `org.gecko.emf.exporter.csv.api`), an unexpected bundle appears in resolution results – unexpected in the sense that it is not specific to this format – along with its transitive dependencies (e.g. `org.gecko.emf.exporter.ods` with `com.github.miachm.sods` in some cases, `org.gecko.emf.exporter.xlsx` with `org.apache.servicemix.bundles.poi` in other cases). Out of 8 `test.bndrun`s, this behavior appears in 3 (`/org.gecko.emf.xlsx.tests/test.bndrun`, `/org.gecko.emf.r_lang.tests/test.bndrun`, `/org.gecko.emf.csv.tests/test.bndrun`) and does not appear in the rest (`/org.gecko.emf.ods.tests/test.bndrun`, `/org.gecko.emf.exporter.xlsx.tests/test.bndrun`, `/org.gecko.emf.exporter.r_lang.tests/test.bndrun`, `/org.gecko.emf.exporter.ods.tests/test.bndrun`, `/org.gecko.emf.exporter.csv.tests/test.bndrun`).

It appears this problem is caused by lack of sufficient information for resolver to determine which specific format of EMF exporter to pull in when detecting OSGi service requirement defined via `org.osgi.service.component.annotations.Reference` – e.g. in `org.gecko.emf.csv.EMFCSVResourceFactoryConfigurator`, `org.gecko.emf.exporter.EMFExporter` is referenced, but even though its target property is correctly specified ("(component.name=EMFCSVExporter)") this information is not passed on to manifest headers generated, i.e. `Require-Capability` manifest header only defines `osgi.service;filter:="(objectClass=org.gecko.emf.exporter.EMFExporter)";effective:=active` which is too generic. 

IMHO it would be useful to understand underlying cause of such non-deterministic behavior, similar non-deterministic behavior mentioned in other question asked on Slack several days ago, which was left without answer (i.e. imports generated only in some bundles, even though all of them use the same pattern in the sense of importing constants class).

![Screenshot from 2023-12-26 02-36-00](https://github.com/geckoprojects-org/org.gecko.emf.utils/assets/9023040/c5fd83be-f352-4aa6-ac6b-0ad6350c0b3e)